app/views/auth/registrations/new.html.haml

```diff
@@ -18,7 +18,7 @@
   .fields-group
     = f.simple_fields_for :account do |ff|
       = ff.input :display_name, wrapper: :with_label, label: false, required: false, input_html: { 'aria-label': t('simple_form.labels.defaults.display_name'), autocomplete: 'off', placeholder: t('simple_form.labels.defaults.display_name') }
-      = ff.input :username, wrapper: :with_label, label: false, required: true, input_html: { 'aria-label': t('simple_form.labels.defaults.username'), autocomplete: 'off', placeholder: t('simple_form.labels.defaults.username'), pattern: '[a-zA-Z0-9_]+', maxlength: 30 }, append: "@#{site_hostname}", hint: false
+      = ff.input :username, wrapper: :with_label, label: false, required: true, input_html: { 'aria-label': t('simple_form.labels.defaults.username'), autocomplete: 'off', placeholder: t('simple_form.labels.defaults.username'), pattern: '[a-zA-Z0-9_]+', maxlength: 30, oninvalid: "this.setCustomValidity('#{t('auth.username_validity')}')", onchange: "try{setCustomValidity('')}catch(e){}", oninput: "setCustomValidity('')" }, append: "@#{site_hostname}", hint: false
```

Comment:
Inline Javascript is not allowed by our Content Security Policy, so this will not work. Instead, have the JS `app/javascript/packs/public.js`.